### PR TITLE
Mappers [PAR-69]

### PR DESCRIPTION
**Pull request by lemniscate8**
Explicitly map data coming from the backend in JSON to frontend data structures. This will allow a some more flexibility in translating from backend to frontend (such as flattening deep objects). Right now these are mainly useful for pinpointing if there are places where the backend isn't returning what's expected.

### CODE REVIEW

File: src/serviceWorker.ts [hunk 1]
```diff
@@ -11,9 +11,9 @@
 // opt-in, read https://bit.ly/CRA-PWA
 
 const isLocalhost = Boolean(
-  window.location.hostname === 'localhost' ||
+  window.location.hostname === "localhost" ||
```
Comment (lemniscate8) [inline src/serviceWorker.ts:14]: Ignore all this, seems I accidentally touched this file when looking for fetches and prettier got carried away.

**Comment by Gatlinn** [comment]
This code looks good to me go ahead and merge when you are ready to

File: src/BookForm.tsx [hunk 2]
```diff
@@ -130,6 +137,7 @@ const BookFormik: React.FC<BookFormProps> = ({
       }}
       initialValues={book}
       onSubmit={(values: Book) => {
+        console.log(values);
```
Comment (Hashiphoto) [inline src/BookForm.tsx]: Not sure if it matters or not, but it may be best not to log

**Review by Hashiphoto — APPROVED** [review]
Looks great. I'll probably revisit the friend search bar after this is merged to use the mapper

File: src/BookForm.tsx (hunk 2 above)
Comment (lemniscate8) [reply src/BookForm.tsx]: Wope! An artifact from testing. I'll remove it.

**Pull request merged**